packages/material-ui/src/ListItem/ListItem.spec.tsx

```diff
@@ -5,17 +5,8 @@ import ListItem from '@material-ui/core/ListItem';
 function BooleanButtonTest() {
   // https://github.com/mui-org/material-ui/issues/14971
 
-  function EditableItemFail(props: { editable: boolean }) {
+  function EditableItemValid(props: { editable?: boolean }) {
     const { editable } = props;
-    // 'boolean' is not assignable to type 'true'
```

Comment:
I should've been explicit that this is the undesired behavior that is fixable with the test case below.